WordPress/Classes/Utility/NotificationRemindersHelper.swift

```diff
@@ -0,0 +1,276 @@
+                    return true
+            }
+
+            return date1 < date2
+        })
+    }
+
+    // TODO: Add method to clear out reminder IDs that are no longer valid.
+
+    /// An array of all reminder identifiers that are currently being tracked.
+    ///
+    private var reminderIds: [String] {
+        let ids = UserDefaults.standard.array(forKey: DefaultsKeys.reminderIds) as? [String]
+        return ids ?? []
+    }
+
+    private enum DefaultsKeys {
+        static let reminderIds = "notification-reminder-ids"
+    }
+
+    enum RequestKeys {
+        static let category = "notification-reminder"
+        static let notificationId = "note_id"
+        static let type = "type"
+        static let typeValue = "local"
+    }
+}
+
+/// Encapsulates the period after which a reminder should be displayed.
+///
+enum NotificationReminderPeriod: CaseIterable {
+    case in20minutes
+    case in1hour
+    case in3hours
+    case tomorrow   // 9am tomorrow
+    case nextWeek   // 9am on the next Monday
+
+    /// A title describing the time period, for display to the user.
+    ///
+    var displayTitle: String {
+        switch self {
+        case .in20minutes:
+            return NSLocalizedString("In 20 Minutes", comment: "Title of a reminder option that sets a reminder in 20 minutes time")
+        case .in1hour:
+            return NSLocalizedString("In 1 Hour", comment: "Title of a reminder option that sets a reminder in 1 hours time")
+        case .in3hours:
+            return NSLocalizedString("In 3 Hours", comment: "Title of a reminder option that sets a reminder in 3 hours time")
+        case .tomorrow:
+            return NSLocalizedString("Tomorrow", comment: "Title of a reminder option that sets a reminder for 9am tomorrow.")
+        case .nextWeek:
+            return NSLocalizedString("Next Week", comment: "Title of a reminder option that sets a reminder for 9am the following Monday.")
+        }
+    }
+
+    /// Date components representing the time period, from the current date.
+    /// Used to schedule a reminder.
+    ///
+    var dateComponents: DateComponents? {
+        return dateComponents(from: Date.today)
+    }
+
+    /// Date components representing the time period, from the specified date.
+    ///
+    func dateComponents(from date: Date) -> DateComponents? {
+        var components = DateComponents()
+
+        switch self {
+        case .in20minutes:
+            components.minute = 20
+            return date.addingComponents(components)?.timeAndDateComponents
+        case .in1hour:
+            components.hour = 1
+            return date.addingComponents(components)?.timeAndDateComponents
+        case .in3hours:
+            components.hour = 3
+            return date.addingComponents(components)?.timeAndDateComponents
+        case .tomorrow:
+            components.day = 1
+            guard let tomorrow = date.addingComponents(components),
```

Comment:
Do you think it will be useful to move `tomorrow` to the private Date extension below like: 
` var tomorrow: Date? {
        var components = DateComponents()
        components.day = 1
        return self.addingComponents(components)
    }`

Also maybe it would be useful to add those to a Date extension that other classes can use?